development/tools/pkg/diskscollector/collector_test.go

```diff
@@ -0,0 +1,198 @@
+package diskscollector
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/kyma-project/test-infra/development/tools/pkg/diskscollector/automock"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	compute "google.golang.org/api/compute/v1"
+)
+
+const regexPattern = "^gke-gkeint.*[-]pvc[-]"
+const sampleDiskName = "gke-gkeint-abc-pvc-xyz"
+
+var (
+	diskNameRegex, _         = regexp.Compile(regexPattern)
+	filterFunc               = NewDiskFilter(diskNameRegex, 1) //age is 1 hour
+	timeNow                  = time.Now()
+	timeNowFormatted         = timeNow.Format(time.RFC3339Nano)
+	timeTwoHoursAgo          = timeNow.Add(time.Duration(-1) * time.Hour)
+	timeTwoHoursAgoFormatted = timeTwoHoursAgo.Format(time.RFC3339Nano)
+	nonEmptyUsers            = []string{"someUser"}
+)
+
+func TestNewDiskFilter(t *testing.T) {
+	//given
+	var testCases = []struct {
+		name                  string
+		expectedFilterValue   bool
+		diskName              string
+		diskCreationTimestamp string
+		diskUsers             []string
+	}{
+		{"Should filter matching disk", true, sampleDiskName, timeTwoHoursAgoFormatted, []string{}},
+		{"Should skip disk with non matching name", false, "otherName", timeTwoHoursAgoFormatted, []string{}},
+		{"Should skip disk recently created", false, sampleDiskName, timeNowFormatted, []string{}},
+		{"Should skip disk with users", false, sampleDiskName, timeTwoHoursAgoFormatted, nonEmptyUsers},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			//when
+			disk := createDisk(testCase.diskName, testCase.diskCreationTimestamp, testCase.diskUsers)
+			collected, err := filterFunc(disk)
+
+			//then
+			require.NoError(t, err)
+			assert.Equal(t, testCase.expectedFilterValue, collected)
+		})
+	}
+
+	t.Run("Should return error on invalid timestamp", func(t *testing.T) {
+
+		//given
+		badTimestamp := "@@@"
+		disk := compute.Disk{
+			CreationTimestamp: badTimestamp,
+		}
+
+		_, err := filterFunc(&disk)
+		assert.Contains(t, err.Error(), fmt.Sprintf("parsing time \"%s\" as", badTimestamp))
+	})
+}
+
+func TestGarbageDiskCollector(t *testing.T) {
+
+	d1 := createDisk(sampleDiskName+"1", timeTwoHoursAgoFormatted, []string{})    //matches removal filter
```

Comment:
Maybe instead of comment name variable more meaningfully? This way even in test scenario below it is easier to understand what's the difference between all those d*.

Reply:
Done.